### PR TITLE
Ehn background windows

### PR DESCRIPTION
**Pull request by pburdet**
Add a method to subtract background to `get_lines_intensity`. The background is average in windows. The windows position is estimated by `estimate_background_windows`. The windows can be plotted by `plot_background_windows`. 

Note
------

Follow up of  #391 
Merge with #296

Example
-----------

```python

>>> bw = spec.estimate_background_windows()
>>> spec.plot_background_windows(bw)
>>> spec.get_lines_intensity(background_windows=bw)

```

### CODE REVIEW

**Comment by to266** [comment]
Cannot fast-forward (please fix the merge conflicts)

**Comment by pburdet** [comment]
Conflict resolve and integration improved

File: hyperspy/_signals/eds.py [hunk 1]
```diff
@@ -704,27 +798,43 @@ def plot(self,
...
+            Each line corresponds to a X-ray lines. In a line, the two first
+            value corresponds to the limit of the left window and the two
+            last values corresponds to the limit of the right window.
+        integration_windows: None or 'auto' or float or 2D array of float
             If not None, add markers at the position of the integration
             windows.
-            If float, use 'estimate_integration_windows`.
+            If 'auto' (or float), the width of the integration windows is 2.0
+            (or float) times the calculated FWHM of the line. see
+            'estimate_integration_windows`.
```
Comment (to266) [inline hyperspy/_signals/eds.py]: Different types of backticks

Comment (pburdet) [inline hyperspy/_signals/eds.py]: Done

**Pull request merged**